deployer/deployer-services/src/main/java/io/hyscale/deployer/services/factory/PodParentFactory.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.deployer.services.factory;
+
+import io.hyscale.deployer.services.handler.PodParentHandler;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class PodParentFactory {
+
+    private List<PodParentHandler> parentHandlerList;
+
+    Map<String, PodParentHandler> kindVsHandlerMap;
+
+    @PostConstruct
+    public void init() {
+        kindVsHandlerMap = new HashMap<>();
+        parentHandlerList.forEach(each -> {
```

Comment:
This will be null right?

Reply:
Yes until we figure out some way to identify parent from pod directly. For now this is a temporary class, will do a TODO instead that makes more sense